src/klipse/plugin.cljs

```diff
@@ -105,24 +107,47 @@
   [element general-settings mode]
   (if-let [opts (@mode-options mode)]
     ;; weird piece of code - see klipsify-with-opts docstring
-    (go (<! ((<! (klipsify-with-opts element general-settings opts mode)))))
+    (go (<! ((<! (klipsify-with-opts element general-settings opts mode #())))))
     (go (js/console.error "cannot find options for mode: " mode ". Supported modes: " (keys @mode-options)))))
 
-(defn ^:export klipsify-no-eval [element general-settings mode]
+(defn ^:export klipsify-no-eval [element general-settings mode on-edit-cb]
   (if-let [opts (@mode-options mode)]
-    (klipsify-with-opts element general-settings opts mode)
+    (klipsify-with-opts element general-settings opts mode on-edit-cb)
     (go #(go (js/console.error "cannot find options for mode: " mode ". Supported modes: " (keys @mode-options))))))
 
 (defn edit-elements [elements general-settings modes]
   (go
-    (loop [elements elements eval-fns []]
-      (if (seq elements)
-        (let [element (first elements)]
-          (if-let [mode (modes element)]
-            (let [eval-fn (<! (klipsify-no-eval element general-settings mode))]
-              (recur (rest elements) (conj eval-fns eval-fn)))
-            (recur (rest elements) eval-fns)))
-        eval-fns))))
+    (let [valid-elements (filter modes elements)
+          ; this channel is posted to whenever an edit is made
+          event-chan (chan (if (general-settings :re_evaluate_all_snippets_on_change)
+                              10
+                              (dropping-buffer 1)))
+          eval-fns (loop [elements valid-elements
+                          eval-fns []
+                          curr-idx 0]
+                     (if (seq elements)
+                       (let [element (first elements)]
+                         (let [eval-fn (<! (klipsify-no-eval element general-settings (modes element) #(go
+                                                                                                         (>! event-chan curr-idx))))]
+                           (recur (rest elements) (conj eval-fns eval-fn) (inc curr-idx))))
+                       eval-fns))]
+      (println eval-fns)
+      ; sub process that reloads other editors
+      (go
+        ; We reevaluate all the other snippets in the order they appear on page
+        ; code once a snippet is edited
+        (loop [idx (<! event-chan)]
+          ; evaluate edited snippet
+          ((eval-fns idx))
+
+          (if (general-settings :re_evaluate_all_snippets_on_change)
```

Comment:
What is the order of evaluation now?
Let's say we have 3 snippets on a page and snippet #2 is edited?

Reply:
I am fine with 2->1->2->3 if it makes the code simpler.
More important, I'd prefer to avoid all risk and leave the code that handles the default case (`:re_evaluate_all_snippets_on_change false`) untouched.